source/fe/fe_values.cc

```diff
@@ -97,6 +97,36 @@ namespace internal
 
     return shape_function_to_row_table;
   }
+
+  namespace
+  {
+    // Check to see if a DoF value is zero, implying that subsequent operations
+    // with the value have no effect.
+    template <typename Number, typename T = void>
+    struct FunctionDoFValues
+    {
+      static bool
+      filter (const Number &value)
```

Comment:
`filter` is an awkward name for the function. Filtering means to selectively choose elements out of a sequence, but here you have a *predicate* that only returns true/false. The struct name `FunctionDoFValues` is also not very helpful. Maybe use `CheckForZero` and `CheckForZero::check()`?